torch/optim/nadam.py

```diff
@@ -268,14 +296,14 @@ def _single_tensor_nadam(params: List[Tensor],
         exp_avg_sq.mul_(beta2).addcmul_(grad, grad, value=1 - beta2)
         denom = exp_avg_sq.div(bias_correction2).sqrt()
 
-        if differentiable:
+        if differentiable or capturable:
             denom = denom.add(eps)
             # Make autograd track the operations
             # by updating the grad and exp_avg directly and not using the
             # scalar "value" argument of addcdiv.
             mu_product_next = mu_product * mu_next
             grad = grad * (-lr * (1. - mu) / (1. - mu_product))
-            exp_avg = grad * (-lr * (1. - mu_next) / (1. - mu_product_next))
+            exp_avg = exp_avg * (-lr * mu_next / (1. - mu_product_next))
```

Comment:
This was actually just incorrect before. It is on my mind to add differentiable correctness tests as a part of the test revamp that I will get to one day.